File: projekty/PO_Projekt3/zolw.py
from zwierze import Zwierze

class Zolw(Zwierze):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 2, 1, 0.25, 1, 0.00, 1.00)

    def odbilAtak(self, atakujacy):
        return atakujacy.getSila() < 5

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Zolw)

    def stworzDziecko(self, cel):
        return Zolw(self._swiat, cel)

    def rysowanie(self):
        return (128, 51, 0)

File: PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Zwierzeta/Zolw.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Zwierzeta;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Zolw extends Zwierze {

	public Zolw(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Zolw.class), Stale.inicjatywa.get(Zolw.class),
				Stale.szansaRuchu.get(Zolw.class), Stale.zasiegRuchu.get(Zolw.class),
				Stale.szansaUcieczki.get(Zolw.class), Stale.szansaRozmnazania.get(Zolw.class));
	}
	
	@Override
	public boolean odbilAtak(Organizm atakujacy) {
		return atakujacy.getSila() < 5;
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Zolw;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Zolw(swiat, cel);
	}

}
